payload/payload.cc

```diff
@@ -10,68 +10,73 @@ namespace sorbet::payload {
-    ENFORCE(gs->classAndModulesUsed() < core::GlobalState::PAYLOAD_MAX_CLASS_AND_MODULE_COUNT,
+        ENFORCE(gs->fieldsUsed() < core::GlobalState::PAYLOAD_MAX_FIELD_COUNT,
+                "Payload defined `{}` fields, which is greater than the expected maximum of `{}`. Consider updating "
+                "`PAYLOAD_MAX_FIELD_COUNT` in `GlobalState`.",
+                gs->fieldsUsed(), core::GlobalState::PAYLOAD_MAX_FIELD_COUNT);
+        ENFORCE(gs->methodsUsed() < core::GlobalState::PAYLOAD_MAX_METHOD_COUNT,
+                "Payload defined `{}` methods, which is greater than the expected maximum of `{}`. Consider updating "
+                "`PAYLOAD_MAX_METHOD_COUNT` in `GlobalState`.",
+                gs->methodsUsed(), core::GlobalState::PAYLOAD_MAX_METHOD_COUNT);
+        ENFORCE(
+            gs->classAndModulesUsed() < core::GlobalState::PAYLOAD_MAX_CLASS_AND_MODULE_COUNT,
             "Payload defined `{}` classes and modules, which is greater than the expected maximum of `{}`. Consider "
             "updating `PAYLOAD_MAX_CLASS_AND_MODULE_COUNT` in `GlobalState`.",
             gs->classAndModulesUsed(), core::GlobalState::PAYLOAD_MAX_CLASS_AND_MODULE_COUNT);
-    ENFORCE(gs->typeMembersUsed() < core::GlobalState::PAYLOAD_MAX_TYPE_MEMBER_COUNT,
+        ENFORCE(
+            gs->typeMembersUsed() < core::GlobalState::PAYLOAD_MAX_TYPE_MEMBER_COUNT,
             "Payload defined `{}` type members, which is greater than the expected maximum of `{}`. Consider updating "
             "`PAYLOAD_MAX_TYPE_MEMBER_COUNT` in `GlobalState`.",
             gs->typeMembersUsed(), core::GlobalState::PAYLOAD_MAX_TYPE_MEMBER_COUNT);
-    ENFORCE(
-        gs->typeArgumentsUsed() < core::GlobalState::PAYLOAD_MAX_TYPE_ARGUMENT_COUNT,
-        "Payload defined `{}` type arguments, which is greater than the expected maximum of `{}`. Consider updating "
-        "`PAYLOAD_MAX_TYPE_ARGUMENT_COUNT` in `GlobalState`.",
-        gs->typeArgumentsUsed(), core::GlobalState::PAYLOAD_MAX_TYPE_ARGUMENT_COUNT);
+        ENFORCE(gs->typeArgumentsUsed() < core::GlobalState::PAYLOAD_MAX_TYPE_ARGUMENT_COUNT,
+                "Payload defined `{}` type arguments, which is greater than the expected maximum of `{}`. Consider "
+                "updating "
+                "`PAYLOAD_MAX_TYPE_ARGUMENT_COUNT` in `GlobalState`.",
+                gs->typeArgumentsUsed(), core::GlobalState::PAYLOAD_MAX_TYPE_ARGUMENT_COUNT);
+    }
+
+    // We can use the kvstore to read in the cached name table. We read this in after the payload has been initialized,
+    // as the cached name table will extend the payload's existing table when the sorbet versions match.
+    if (kvstore) {
+        ENFORCE(!options.noStdlib, "--no-stdlib and --cache-dir are incompatible");
+        auto maybeGsBytes = kvstore->read(core::serialize::Serializer::GLOBAL_STATE_KEY);
+        if (maybeGsBytes.data != nullptr) {
+            Timer timeit(gs->tracer(), "read_global_state.kvstore");
```

Comment:
we can consider whether we want to change this to `read_name_table.kvstore`

Reply:
I think that's reasonable. I don't think we have any charts that rely on this name, and it'll make more sense in traces.